pkg/helpers/helpers.go

```diff
@@ -189,9 +185,5 @@ func IsBpfAppStateConditionFailure(conditions *[]metav1.Condition) bool {
 		log.Info("more than one condition found", "numConditions", numConditions)
 	}
 
-	if (*conditions)[0].Type == string(bpfmaniov1alpha1.BpfAppCondSuccess) {
-		return false
-	} else {
-		return true
-	}
+	return (*conditions)[0].Type == string(bpfmaniov1alpha1.BpfAppCondPending)
```

Comment:
same here does [0] gruanteed to be in pending state or we better off loop through all and return the 1st one in pending state ?